lib/src/main/java/com/dynatrace/file/util/PollBasedFilePoller.java

```diff
@@ -59,21 +74,86 @@ public boolean fileContentsUpdated() {
     return changedSinceLastInquiry.getAndSet(false);
   }
 
+  private synchronized void initialPoll() {
+    try {
+      prevChecksum = getFileChecksum();
+      prevModMillis =
+          Files.getLastModifiedTime(absoluteFilePath, LinkOption.NOFOLLOW_LINKS).toMillis();
+    } catch (IOException e) {
+      logger.warning(
+          () ->
+              String.format(
+                  "Failed to read file %s, stopping polling. Error: %s", absoluteFilePath, e));
+      worker.cancel(true);
```

Comment:
Would the poller and the caller be able to recover if the file was added later or permissions were changed? If so, we should keep polling (same below).

Reply:
Changed it so this is true now.